tests/validation/tests/v3_api/test_cli.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pytest
+
+from .cli_common import RancherCli, DEFAULT_CONTEXT
+from .common import CATTLE_TEST_URL, USER_TOKEN, get_user_client
+from .cli_objects import CliProject
+
+CLUSTER_NAMES = [os.environ.get("RANCHER_CLUSTER_NAME_1", ""),
+                 os.environ.get("RANCHER_CLUSTER_NAME_2", "")]
+
+
+def test_list_clusters(rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Listing Clusters")
+    clusters = rancher_cli.get_clusters()
+    assert len(clusters) == 2
+    for cluster in clusters:
+        assert cluster["name"] in CLUSTER_NAMES
+
+
+def test_context_switching(rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Context Switching")
+    clusters = rancher_cli.get_clusters()
+    client = get_user_client()
+    projects = client.list_project()
```

Comment:
I think we have to use the api commands in this case. The problem is there is no way (that I can find) to retrieve project IDs outside of the current cluster context, and we cannot switch context based on Name like:
```
$ rancher context switch c-88rmh:System
FATA[0000] Not found: c-88rmh:System                    
```

I think there is one way we can do it. When we run `rancher context switch $PROJECT_NAME`, it will either switch to the project if possible (it is the only project with that name among all of your clusters) or provide an error like: `FATA[0000] Multiple resources of type project found for name System: [c-88rmh:p-jd7cr c-qcqcs:p-8bzdh]`. If we get the error, we could parse the result list in the brackets and do what you've suggested for that list.

For the case of `get_current_projects` or this specific test, we would say `rancher context switch System`. The only danger is if the "System" Project is somehow deleted, then we could get unexpected results.